test/blackbox/common/BlackboxTestsNetworkConf.cpp

```diff
@@ -296,6 +296,191 @@ TEST_P(NetworkConfig, PubSubInterfaceWhitelistUnicast)
     reader.block_for_all();
 }
 
+// Setting the interface whitelist by the interface names,
+// check if the connection is established anyways.
+// All available interfaces case for both publisher and subscriber
+TEST_P(NetworkConfig, PubSubInterfaceWhitelistName)
+{
+    PubSubReader<HelloWorldPubSubType> reader(TEST_TOPIC_NAME);
+    PubSubWriter<HelloWorldPubSubType> writer(TEST_TOPIC_NAME);
+
+    std::vector<IPFinder::info_IP> interfaces;
+    use_udpv4 ? GetIP4s(interfaces) : GetIP6s(interfaces);
+
+
+    for (const auto& interface : interfaces)
+    {
+        descriptor_->interfaceWhiteList.push_back(interface.dev);
+    }
+
+    reader.reliability(eprosima::fastrtps::RELIABLE_RELIABILITY_QOS).history_depth(10).
+            disable_builtin_transport().
+            add_user_transport_to_pparams(descriptor_).init();
+
+    ASSERT_TRUE(reader.isInitialized());
+
+    writer.reliability(eprosima::fastrtps::RELIABLE_RELIABILITY_QOS).history_depth(10).
+            disable_builtin_transport().
+            add_user_transport_to_pparams(descriptor_).init();
+
+    ASSERT_TRUE(writer.isInitialized());
+
+    // Because its volatile the durability
+    // Wait for discovery.
+    writer.wait_discovery();
+    reader.wait_discovery();
+
+    auto data = default_helloworld_data_generator();
+
+    reader.startReception(data);
+
+    writer.send(data);
+    ASSERT_TRUE(data.empty());
+    reader.block_for_all();
+
+}
+
+// Allow to configure different whitelist for pub and sub using interface names
+void asymmetric_whitelist_name_test(
```

Comment:
I suggest to add a boolean parameter to `asymmetric_whitelist_test`, to indicate if it should add the address or the name (e.g. interface_name, false by default).

I would also modify its name to `interface_whitelist_test`, so you could also use it for the all interfaces in both endpoints case.

Reply:
Isn't it possible to use this method for all new added tests?